### PR TITLE
initial server that interfaces with USB stick

### PR DESCRIPTION
**Pull request by benadida**
lots of boilerplate for typescript etc, a simple server that execs out commands to check USB status and eject when needed.

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+//
+// Just the HTTP glue to the functionality, no implementations.
+// All actual implementations are in scanner.ts
```
Comment (beausmith) [inline src/server.ts]: copy/pasta :)

File: src/server.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    if (blockDevice.children) {
+      const childResult = findBlockDevice(blockDevice.children)
+      if (childResult) {
+        return childResult
+      }
+    }
+  }
+
+  // eslint-disable-next-line no-null/no-null
+  return null
```
Comment (beausmith) [inline src/server.ts]: Why not simply `return` without the `null` and save yourself having to disable-eslint elsewhere in the code where `undefined` should work just fine?

I'd test myself, but I assume that I need to have the production hardware, no?

**Comment by beausmith** [comment]
Generally use of `null` is not recommended in TypeScript: `#reading` 😉 
- https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined
- https://basarat.gitbooks.io/typescript/docs/javascript/null-undefined.html
- https://www.youtube.com/watch?v=PSGEjv3Tqo0&feature=youtu.be&t=9m21s

File: src/server.ts (hunk 1 above)
Comment (benadida) [reply src/server.ts]: indeed, I will need to do a cleanup.

File: src/server.ts (hunk 2 above)
Comment (benadida) [reply src/server.ts]: I'm on board with not returning null, I just got to a point where it was working and wanted to ship it.

**Comment by benadida** [comment]
filing #2 and #3 to track needed changes.

**Pull request merged**